src/main/java/testapp/adapter/memory/InMemoryUserRepository.java

```diff
@@ -26,9 +25,9 @@ public InMemoryUserRepository() {
   }
 
   @Override
-  public User findUser(Integer userId) {
+  public User findUser(Integer facultyNumber) {
     for (User user : userList) {
-      if (user.getId().equals(userId)) {
+      if (user.getFacultyNumber().equals(facultyNumber)) {
```

Comment:
Actually the requirement didn't even specify if it would be a university, you might want to read the definition of the word student